crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::{
+    runtime_api::api,
+    submit_extrinsic,
+    Client,
+    CodeHash,
+    ContractMessageTranscoder,
+    DefaultConfig,
+    ExtrinsicOpts,
+    PairSigner,
+    TokenMetadata,
+};
+use crate::{
+    cmd::extrinsics::{
+        events::DisplayEvents,
+        parse_code_hash,
+        ErrorVariant,
+    },
+    name_value_println,
+};
+use anyhow::{
+    Ok,
+    Result,
+};
+use std::fmt::Debug;
+use subxt::{
+    Config,
+    OnlineClient,
+};
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "remove", about = "Remove a contract's code")]
+pub struct RemoveCommand {
+    /// The hash of the smart contract code already uploaded to the chain.
+    #[clap(long, value_parser = parse_code_hash)]
+    code_hash: Option<<DefaultConfig as Config>::Hash>,
+    #[clap(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl RemoveCommand {
+    pub fn is_json(&self) -> bool {
+        self.output_json
+    }
+
+    pub fn run(&self) -> Result<(), ErrorVariant> {
+        let artifacts = self.extrinsic_opts.contract_artifacts()?;
+        let transcoder = artifacts.contract_transcoder()?;
+        let signer = super::pair_signer(self.extrinsic_opts.signer()?);
+
+        let artifacts_path = artifacts.artifact_path().to_path_buf();
+
+        let final_code_hash = match (self.code_hash.as_ref(), artifacts.code.as_ref()) {
+            (Some(code_h), Some(_)) => {
+                Ok(code_h.0)
+            }
+            (Some(code_h), None) => {
+                Ok(code_h.0)
+            }
+            (None, Some(_)) => {
+                let interm_artifacts_code = artifacts.code_hash()?;
+                Ok(interm_artifacts_code)
+            }
+            (None, None) => {
+                Err(
+                    anyhow::anyhow!(
+                    "No code_hash was provided or contract code was not found from artifact file {}. Please provide a code hash with --code-hash argument or specify the path for artifacts files with --manifest-path", 
+                    artifacts_path.display()
+                ))
+            }
+        };
```

Comment:
```suggestion
        let final_code_hash = match (self.code_hash.as_ref(), artifacts.code.as_ref()) {
            (Some(code_h), _) => {
                Ok(code_h.0)
            }
            (None, Some(_)) => {
                artifacts.code_hash()
            }
            (None, None) => {
                Err(
                    anyhow::anyhow!(
                    "No code_hash was provided or contract code was not found from artifact file {}. Please provide a code hash with --code-hash argument or specify the path for artifacts files with --manifest-path", 
                    artifacts_path.display()
                ))
            }
        }?;
```
Then remove the `? from `final_code_hash` below

Reply:
Thanks it seems far better